GW2EIEvtcParser/EncounterLogic/Fractals/SilentSurf/Kanaxai.cs

```diff
@@ -359,6 +378,55 @@ internal override void ComputeNPCCombatReplayActors(NPC target, ParsedEvtcLog lo
                         int end = (int)cast.ExpectedEndTime; // actual end is often much later, just use expected end for short highlight
                         replay.Decorations.Add(new CircleDecoration(false, 0, 180, (start, end), "rgba(0, 100, 255, 0.5)", new AgentConnector(target), 20));
                     }
+                    // Dread Visage
+                    var dreadVisageAspects = casts.Where(x => x.SkillId == DreadVisageAspectSkill).ToList();
+                    bool hasSugarRush = false;
+
+                    // Check if the log contains Sugar Rush
+                    foreach (Buff fractalInstability in log.Buffs.BuffsBySource[Source.FractalInstability])
+                    {
+                        if (log.CombatData.GetBuffData(fractalInstability.ID).Any(x => x.To.IsPlayer) && fractalInstability.ID == MistlockInstabilitySugarRush)
```

Comment:
Isn't that simply hasSugarRush = log.CombatData.GetBuffData(MistlockInstanbilitySugarRush).Any(x => x.To.IsPlayer) ?

Reply:
yeah that's smarter than what i did